src/modules/emote_autocomplete/components/Emotes.jsx

```diff
@@ -0,0 +1,92 @@
+import React, {useEffect, useMemo} from 'react';
+import {Button} from 'rsuite';
+import useChatInput from '../hooks/ChatInput.jsx';
+import Emote from '../../../common/components/Emote.jsx';
+import styles from './Emotes.module.css';
+import useRowNavigation from '../hooks/RowKeyboardNavigation.jsx';
+import {isEmoteAutocompletable} from '../../../utils/autocomplete.js';
+import keyCodes from '../../../utils/keycodes.js';
+
+const MAX_EMOTES_SHOWN = 5;
+const EMOTE_ROW_HEIGHT = 36;
+const BOTTOM_OFFSET = 120; // height of everything below the chat window
+
+function calcMaxHeight() {
+  const canShow = (window.innerHeight - BOTTOM_OFFSET) / EMOTE_ROW_HEIGHT;
+
+  if (canShow >= MAX_EMOTES_SHOWN) {
+    return MAX_EMOTES_SHOWN;
+  }
+
+  return canShow;
+}
+
+const shortenName = (displayName) => displayName.split(' ')[0];
+
+let navigationCallback;
+function setNavigationCallback(callback) {
+  navigationCallback = callback;
+}
+
+export default function Emotes({chatInputElement, repositionPopover, autocomplete}) {
+  const [emotes, setEmotes] = useChatInput(chatInputElement);
+  const shortEmotes = useMemo(() => emotes.slice(0, calcMaxHeight()), [emotes]);
```

Comment:
what happens when a resize occurs? do we care?

Reply:
do you think the user will resize the window whilst writing a message with autocomplete open?